File: frontend/src/types/project.ts
// TypeScript types matching Go backend models

import type { Asset } from './canvas';
import type { NumberSettings, ManualNumber, NumberItem } from './numbering';
import type { SheetSettings, CropBleedSettings, PrintSettings } from './sheet';
import type { Layer } from './canvas';

export type ProjectType = 'receipt' | 'foil';

export interface Project {
  id: string;
  name: string;
  description: string;
  type: ProjectType;
  imagePath: string;
  thumbnailPath: string;
  createdAt: string;
  updatedAt: string;
}

export interface ProjectFull {
  project: Project;
  assets: Asset[];
  numberSettings: NumberSettings | null;
  manualNumbers: ManualNumber[];
  numberItems: NumberItem[];
  sheetSettings: SheetSettings | null;
  cropBleedSettings: CropBleedSettings | null;
  printSettings: PrintSettings | null;
  layers: Layer[];
}

File: backend/model/project.go
package model

import "time"

// ProjectType distinguishes between the two editing systems.
type ProjectType string

const (
	ProjectTypeReceipt ProjectType = "receipt"
	ProjectTypeFoil    ProjectType = "foil"
)

// Project represents a Recipta project containing all layout, numbering, and print configuration.
type Project struct {
	ID            string      `json:"id"`
	Name          string      `json:"name"`
	Description   string      `json:"description"`
	Type          ProjectType `json:"type"`
	ImagePath     string      `json:"imagePath"`
	ThumbnailPath string      `json:"thumbnailPath"`
	CreatedAt     time.Time   `json:"createdAt"`
	UpdatedAt     time.Time   `json:"updatedAt"`
}

// ProjectFull contains a project with all its related settings loaded.
type ProjectFull struct {
	Project       *Project           `json:"project"`
	Assets        []*Asset           `json:"assets"`
	NumberSetting *NumberSettings    `json:"numberSettings"`
	ManualNumbers []*ManualNumber    `json:"manualNumbers"`
	NumberItems   []*NumberItem      `json:"numberItems"`
	SheetSetting  *SheetSettings     `json:"sheetSettings"`
	CropBleed     *CropBleedSettings `json:"cropBleedSettings"`
	PrintSetting  *PrintSettings     `json:"printSettings"`
	Layers        []*Layer           `json:"layers"`
}
